fairscale/experimental/wgit/repo.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pathlib
+from pathlib import Path
+import sys
+from typing import Union
+
+from .pygit import PyGit
+from .sha1_store import SHA1_store
+
+
+class Repo:
+    def __init__(self, parent_dir: Path, init: bool = False) -> None:
+        """Features:
+        1. Create the wgit directory if it does not exist.
+        2. SHA1Store.init()
+        3. Create SHA1 .wgit/sha1_refs.json
+        3. Initialize a .git directory within the .wgit using `git init`.
+        4. add a .gitignore within the .wgit directory, so that the git repo within will ignore `sha1_refs.json`
+        """
+        # If repo does not exist, creates a new wgit repo object with self.repo.path pointing to the path of repo
+        # and notes all the internal files.
+        # else, if repo already exists: create a pygit object from the .wgit/.git.
+        self.wgit_parent = parent_dir
+        self._repo_path: Union[None, Path] = None
+        self._wgit_dir = Path(".wgit")
+        self._metadata_file = Path(".wgit/checkpoint.pt")
+        self._sha1_ref = Path(".wgit/sha1_refs.json")
```

Comment:
should this be part of sha1 store class? It is for sha1 store for ref counting for delete, right?